core/testcontainers/core/container.py

```diff
@@ -86,8 +93,49 @@ def start(self) -> Self:
             logger.debug("Creating Ryuk container")
             Reaper.get_instance()
         logger.info("Pulling image %s", self.image)
-        docker_client = self.get_docker_client()
         self._configure()
+
+        # container hash consisting of run arguments
+        args = (
+            self.image,
+            self._command,
+            self.env,
+            self.ports,
+            self._name,
+            self.volumes,
+            str(tuple(sorted(self._kwargs.items()))),
+        )
+        hash_ = hashlib.sha256(bytes(str(args), encoding="utf-8")).hexdigest()
```

Comment:
Ideally we use the *full* container create request as the hash input. In tc-java, this is the `CreateContainerCmd` from docker-java, I guess we have some equivalent request object from the Docker Python SDK somewhere available?

Reply:
Yeah, that would be the ideal solution. Unfortunately, I could not find an equivalent function to `CreateContainerCmd` in the Docker SDK for Python: Not by going through the documentation and not by browsing the code base.